kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -360,3 +362,58 @@ def accounts(accts: list[KInner]) -> KInner:
             else:
                 wrapped_accounts.append(acct)
         return build_assoc(KApply('.AccountCellMap'), KLabel('_AccountCellMap_'), wrapped_accounts)
+
+    def pretty_print(self, kast: KAst, *, unalias: bool = True) -> str:
+        if unalias and isinstance(kast, KInner):
+            kast = undo_aliases(self.definition, kast)
+        if isinstance(kast, KInner):
+            kast = KEVM._int_token_to_hex(kast)
+        return super().pretty_print(kast, unalias=False)
+
+    @staticmethod
+    def _int_token_to_hex(kast: KInner) -> KInner:
+        def _helper(_kast: KInner) -> KInner:
+            if type(_kast) is KToken and _kast.sort == INT:
+                return KToken(hex(int(_kast.token)), INT)
+            if type(_kast) is KToken and _kast.sort == BYTES:
+                return KToken('0x' + pretty_bytes(_kast).hex(), BYTES)
+            return _kast
+
+        conjuncts = flatten_label('#And', kast)
+        term = None
+        for c in conjuncts:
+            if type(c) is KApply and c.is_cell:
+                if term:
+                    raise ValueError(f'Found two configurations in pattern:\n\n{term}\n\nand\n\n{c}')
+                term = c
+                for cell in [
+                    'K_CELL',
+                    'OUTPUT_CELL',
+                    'PROGRAM_CELL',
+                    'ID_CELL',
+                    'CALLER_CELL',
+                    'CALLDATA_CELL',
+                    'CALLVALUE_CELL',
+                    'WORDSTACK_CELL',
+                    'LOCALMEM_CELL',
+                    'ORIGIN_CELL',
+                    'LOGSBLOOM_CELL',
+                    'EXTRADATA_CELL',
+                    'ACCOUNTS_CELL',
+                    'MESSAGES_CELL',
+                    'PREVCALLER_CELL',
+                    'PREVORIGIN_CELL',
+                    'NEWCALLER_CELL',
+                    'NEWORIGIN_CELL',
+                    'EXPECTEDREASON_CELL',
+                    'EXPECTEDADDRESS_CELL',
+                    'EXPECTEDVALUE_CELL',
+                    'EXPECTEDDATA_CELL',
+                    'EXPECTEDEVENTADDRESS_CELL',
+                    'ACTIVEACCOUNTS_CELL',
+                ]:
+                    try:
+                        kast = set_cell(kast, cell, bottom_up(_helper, get_cell(kast, cell)))
```

Comment:
I think this can be simplified into a single top-down traversal using `map_inner` (that should improve performance too)

You need two helper functions:

* `match_cell_name`: checks if the term is a cell with the name in the list. If yes, recur with `rewrite_bytes_to_hex`, otherwise, recur with `match_cell_name`.
* `rewrite_bytes_to_hex`: this is the same as `_helper`, basically.

Reply:
So my understanding is that the goal is to rewrite all `Int` and `Bytes` tokens to something, but only if it is a descendant of one of the predefined cells.

In this case I think it is cleaner and more efficient to explicitly do the recursion, like

```python3
CELLS_TO_REWRITE = { ... }

def _int_token_to_hex(kast: KInner) -> KInner:
    def match_cells(kast: KInner) -> KInner:
        if isinstance(kast, KApply) and kast.is_cell and kast.label.name in CELLS_TO_REWRITE:
            return kast.map_inner(rewrite_bytes)
        return kast.map_inner(match_cells)

    def rewrite_bytes(kast: KInner) -> KInner:
        if type(kast) is KToken and kast.sort == INT:
            return KToken(hex(int(kast.token)), INT)
        if type(kast) is KToken and kast.sort == BYTES:
            return KToken('0x' + pretty_bytes(kast).hex(), BYTES)
        return kast.map_inner(rewrite_bytes)  # same as _helper, except for this line

    return kast.map_inner(match_cells)
```

(I did not test this.)

Reply:
What was the resolution of this thread?